<a href="https://colab.research.google.com/github/reo113/GuideHouse_Project/blob/elena-branch/GuideHouse_Colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')
!pip install -q xlrd


Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
import os
import re
import datetime
from sklearn.preprocessing import LabelEncoder
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns


In [3]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score

In [4]:
!git clone https://github.com/reo113/GuideHouse_Project

Cloning into 'GuideHouse_Project'...
remote: Enumerating objects: 141, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (30/30), done.
remote: Total 141 (delta 15), reused 5 (delta 2), pack-reused 109
Receiving objects: 100% (141/141), 2.00 MiB | 20.52 MiB/s, done.
Resolving deltas: 100% (59/59), done.


In [5]:
%load GuideHouse_Project/guidehouse.ipynb

In [6]:
def getDate(filename):
    match_str = re.search(r'\d{4}-\d{2}-\d{2}',filename)
    try:
        return datetime.datetime.strptime(match_str.group(), '%Y-%m-%d').date()
    except ValueError:
        return None


In [7]:
def cleanColumn(name):
    name = name.upper()
    clean_name = re.sub(r'[^a-zA-Z0-9_]', '', name)
    clean_name = clean_name.replace('\n', ' ')
    if clean_name == 'SHELTERNFI':
        clean_name = 'SHELTER'
    if clean_name == 'PROTECTIONCP':
        clean_name = 'PC_CP'
    if clean_name == 'PROTECTIONGBV':
        clean_name = 'PC_GBV'
    if clean_name == 'PROTECTIONMA':
        clean_name = 'PC_MA'
    if clean_name == 'TOTALREACHED':
        clean_name = 'PEOPLEREACHED'
    return clean_name

In [8]:
#using csv and xlsx
#changed csv_dir
csv_dir = '/content/GuideHouse_Project/ukraine_data_excel'


In [9]:
#using xlsx
sheet1_df = []
sheet2_df = []
sheet3_df = []

In [10]:


count=0
for filename in os.listdir(csv_dir):

  print(filename)

  if filename.endswith('.xlsx'):
    file_path = os.path.join(csv_dir,filename)
    excel_df = pd.ExcelFile(file_path)
    files_date = getDate(filename)
    for sheet in excel_df.sheet_names:
      df = pd.read_excel(excel_df,sheet_name=sheet)
      df.columns = [cleanColumn(col) for col in df.columns]
      df['DATE'] = files_date
      if sheet == 'Num_of_Orgs_by_Oblast':
        sheet1_df.append(df)
      elif sheet == 'People_Reached_by_Oblast':
        sheet2_df.append(df)
      elif sheet == 'UDE_Inputs':
        sheet3_df.append(df)



Data_ Round 37 - Ukraine 5W - 2023-01-05.xlsx
Data_ Round 14 - Ukraine 5W - 2022-06-09.xlsx
Data_ Round 31 - Ukraine 5W - 2022-10-13.xlsx
Data_ Round 18 - Ukraine 5W - 2022-07-07.xlsx
Data_ Round 35 - Ukraine 5W - 2022-12-08.xlsx
Data_ Round 10 - Ukraine 5W - 2022-05-12.xlsx
Data_ Round 8 - Ukraine 5W - 2022-04-28.xlsx
Data_ Round 28 - Ukraine 5W - 2022-09-22.xlsx
Data_ Round 29 - Ukraine 5W - 2022-09-29.xlsx
Data_ Round 21 - Ukraine 5W - 2022-07-28.xlsx
Data_ Round 9 - Ukraine 5W - 2022-05-06.xlsx
Data_ Round 23 - Ukraine 5W - 2022-08-11.xlsx
Data_ Round 33 - Ukraine 5W - 2022-11-10.xlsx
Data_ Round 25 - Ukraine 5W - 2022-08-25.xlsx
Data_ Round 30 - Ukraine 5W - 2022-09-29.xlsx
Data_ Round 13 - Ukraine 5W - 2022-06-02.xlsx
Data_ Round 17 - Ukraine 5W - 2022-06-30.xlsx
Data_ Round 36 - Ukraine 5W - 2022-12-22.xlsx
Data_ Round 26 - Ukraine 5W - 2022-09-08.xlsx
Data_ Round 19 - Ukraine 5W - 2022-07-14.xlsx
Data_ Round 12 - Ukraine 5W - 2022-05-26.xlsx
Data_ Round 15 - Ukraine 5W - 2022-0

In [11]:
combined_sheet1 = pd.concat(sheet1_df, keys=[str(df['DATE'].iloc[0]) for df in sheet1_df])
res_sheet1 = combined_sheet1.groupby(['ADMIN1_ID', 'DATE']).sum().reset_index()
merged_sheet1 = pd.merge(res_sheet1, combined_sheet1[['ADMIN1_ID', 'DATE', 'OBLAST']], on=['ADMIN1_ID', 'DATE'], how='left')


<ipython-input-11-f126ab6578e9>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  res_sheet1 = combined_sheet1.groupby(['ADMIN1_ID', 'DATE']).sum().reset_index()


In [12]:
combined_sheet2 = pd.concat(sheet2_df, keys=[str(df['DATE'].iloc[0]) for df in sheet2_df])
res_sheet2 = combined_sheet2.groupby(['ADMIN1_ID', 'DATE']).sum().reset_index()
merged_sheet2 = pd.merge(res_sheet2, combined_sheet2[['ADMIN1_ID', 'DATE', 'OBLAST']], on=['ADMIN1_ID', 'DATE'], how='left')


<ipython-input-12-86453b5296f8>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  res_sheet2 = combined_sheet2.groupby(['ADMIN1_ID', 'DATE']).sum().reset_index()


In [13]:
combined_sheet3 = pd.concat(sheet3_df, keys=[str(df['DATE'].iloc[0]) for df in sheet3_df])
res_sheet3 = combined_sheet3.groupby(['ADMIN1_ID', 'DATE']).sum().reset_index()
merged_sheet3 = pd.merge(res_sheet3, combined_sheet3[['ADMIN1_ID', 'DATE', 'OBLAST']], on=['ADMIN1_ID', 'DATE'], how='left')


<ipython-input-13-b2b5d3c940f8>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  res_sheet3 = combined_sheet3.groupby(['ADMIN1_ID', 'DATE']).sum().reset_index()


In [14]:
merged_sheet1.shape

(750, 20)

In [15]:
merged_sheet3.shape


(750, 5)

In [16]:
merged_sheet1.iloc[26]


ADMIN1_ID                UA05
DATE               2022-11-25
CCCM                      0.0
CCS                       0.0
ETC                       0.0
EDUCATION                 0.0
FSL                       0.0
HEALTH                    0.0
LOGISTICS                 0.0
MPC                       0.0
NUTRITION                 0.0
SHELTER                   0.0
PC_CP                     0.0
PC_GBV                    0.0
PC_MA                     0.0
WASH                      0.0
TOTAL                     0.0
PROTECTIONTOTAL           0.0
PC_PC                     0.0
OBLAST              Vinnytska
Name: 26, dtype: object

In [17]:
#Num_of_Orgs_by_Oblast
merged_sheet1.iloc[20:35]


ADMIN1_ID        DATE  CCCM  CCS  ETC  EDUCATION   FSL  HEALTH  LOGISTICS  \
20      UA05  2022-09-22   2.0  1.0  0.0        4.0  20.0    13.0        0.0   
21      UA05  2022-09-29   6.0  2.0  0.0       20.0  68.0    48.0        0.0   
22      UA05  2022-09-29   6.0  2.0  0.0       20.0  68.0    48.0        0.0   
23      UA05  2022-10-13   3.0  1.0  0.0       11.0  35.0    24.0        0.0   
24      UA05  2022-10-27   3.0  1.0  0.0       13.0  36.0    24.0        0.0   
25      UA05  2022-11-10   0.0  0.0  0.0        0.0   0.0     0.0        0.0   
26      UA05  2022-11-25   0.0  0.0  0.0        0.0   0.0     0.0        0.0   
27      UA05  2022-12-08   3.0  1.0  0.0       13.0  36.0    27.0        0.0   
28      UA05  2022-12-22   2.0  1.0  0.0       13.0  36.0    33.0        0.0   
29      UA05  2023-01-05   3.0  1.0  1.0       17.0  39.0    33.0        0.0   
30      UA07  2022-04-28   1.0  0.0  0.0        1.0   5.0     6.0        0.0   
31      UA07  2022-05-06   1.0  0.0  0.0        1.0   6.0     7.0        0.0   
32      UA07  2022-05-12   1.0  0.0  0.0        1.0   8.0     7.0        0.0   
33      UA07  2022-05-19   1.0  0.0  0.0        2.0   9.0     8.0        0.0   
34      UA07  2022-05-26   0.0  0.0  0.0        2.0   9.0     7.0        0.0   

     MPC  NUTRITION  SHELTER  PC_CP  PC_GBV  PC_MA  WASH  TOTAL  \
20   6.0        0.0      7.0    4.0     3.0    3.0   9.0   58.0   
21  34.0        0.0     46.0   36.0    14.0    8.0  30.0  214.0   
22  34.0        0.0     46.0   36.0    14.0    8.0  30.0  214.0   
23  17.0        0.0     23.0   25.0    16.0    5.0  13.0  123.0   
24  17.0        0.0     23.0   25.0    10.0    5.0  13.0  119.0   
25   0.0        0.0      0.0    0.0     0.0    0.0   0.0    0.0   
26   0.0        0.0      0.0    0.0     0.0    0.0   0.0    0.0   
27  18.0        0.0     24.0   26.0    11.0    7.0  13.0  125.0   
28  18.0        0.0     24.0   29.0    11.0    8.0  13.0  130.0   
29  19.0        0.0     25.0   33.0    10.0    8.0  14.0  138.0   
30   8.0        1.0      4.0    3.0     3.0    2.0   4.0   22.0   
31   7.0        1.0      4.0    4.0     4.0    2.0   4.0   23.0   
32   8.0        1.0      6.0    6.0     4.0    2.0   5.0   29.0   
33   8.0        1.0      6.0    6.0     4.0    2.0   5.0   30.0   
34   7.0        1.0      7.0    7.0     4.0    4.0   5.0   31.0   

    PROTECTIONTOTAL  PC_PC     OBLAST  
20              0.0    0.0  Vinnytska  
21              0.0    0.0  Vinnytska  
22              0.0    0.0  Vinnytska  
23              0.0    0.0  Vinnytska  
24              0.0    0.0  Vinnytska  
25              0.0    0.0  Vinnytska  
26              0.0    0.0  Vinnytska  
27              0.0    0.0  Vinnytska  
28              0.0    0.0  Vinnytska  
29              0.0    0.0  Vinnytska  
30             11.0    6.0   Volynska  
31              0.0    6.0   Volynska  
32             15.0    9.0   Volynska  
33             15.0    9.0   Volynska  
34             17.0    7.0   Volynska

In [18]:
merged_sheet3.iloc[20:35]


ADMIN1_ID        DATE  PEOPLEREACHED  NUMBEROFORGANISATIONS     OBLAST
20      UA05  2022-09-22         395394                     58  Vinnytska
21      UA05  2022-09-29         790788                    214  Vinnytska
22      UA05  2022-09-29         790788                    214  Vinnytska
23      UA05  2022-10-13         395394                    123  Vinnytska
24      UA05  2022-10-27         395394                    119  Vinnytska
25      UA05  2022-11-10         449220                    125  Vinnytska
26      UA05  2022-11-25         449220                    125  Vinnytska
27      UA05  2022-12-08         449220                    125  Vinnytska
28      UA05  2022-12-22         449220                    130  Vinnytska
29      UA05  2023-01-05         449220                    138  Vinnytska
30      UA07  2022-04-28          85800                     22   Volynska
31      UA07  2022-05-06          85800                     23   Volynska
32      UA07  2022-05-12          85900                     29   Volynska
33      UA07  2022-05-19          86900                     30   Volynska
34      UA07  2022-05-26          86900                     31   Volynska

In [19]:
merged_sheet1['OBLAST'].value_counts()


Vinnytska           30
Odeska              30
Chernihivska        30
Chernivetska        30
Cherkaska           30
Khmelnytska         30
Khersonska          30
Kharkivska          30
Ternopilska         30
Sumska              30
Rivnenska           30
Poltavska           30
Mykolaivska         30
Volynska            30
Lvivska             30
Luhanska            30
Kirovohradska       30
Kyivska             30
Ivano-Frankivska    30
Zaporizka           30
Zakarpatska         30
Zhytomyrska         30
Donetska            30
Dnipropetrovska     30
Kyiv                30
Name: OBLAST, dtype: int64

In [20]:
sum(merged_sheet3['PEOPLEREACHED'].value_counts())


750

In [21]:
# replace zero values in the empty tuple to NAN for future linear interpolation
#NOTE : THIS IS HARD CODED NOT OPTIMAL
merged_sheet1.iloc[25] = merged_sheet1.iloc[25].replace(0,np.nan)
merged_sheet1.iloc[26] = merged_sheet1.iloc[26].replace(0,np.nan)

merged_sheet3.iloc[25] = merged_sheet3.iloc[25].replace(merged_sheet3.iloc[25]['PEOPLEREACHED'],np.nan)
merged_sheet3.iloc[26] = merged_sheet3.iloc[26].replace(merged_sheet3.iloc[26]['PEOPLEREACHED'],np.nan)


In [22]:
merged_sheet1['PEOPLE_REACHED'] = merged_sheet3['PEOPLEREACHED']


In [23]:
#check for null values
merged_sheet1.isna().any().any()

True

In [24]:
# check for null values in specific row
merged_sheet1.iloc[26].isna().any()


True

In [25]:
#filled null values using linear interpolation
interpolated_df = merged_sheet1.interpolate(method='linear')


In [26]:
interpolated_df.isna().any().any()


False

In [27]:
# rounded all int and float dtypes since interpolation caused floats
interpolated_df = interpolated_df.applymap(lambda x: round(x, 1) if isinstance(x, (int, float)) else x)


In [28]:
interpolated_df.iloc[20:35]


ADMIN1_ID        DATE  CCCM  CCS  ETC  EDUCATION   FSL  HEALTH  LOGISTICS  \
20      UA05  2022-09-22   2.0  1.0  0.0        4.0  20.0    13.0        0.0   
21      UA05  2022-09-29   6.0  2.0  0.0       20.0  68.0    48.0        0.0   
22      UA05  2022-09-29   6.0  2.0  0.0       20.0  68.0    48.0        0.0   
23      UA05  2022-10-13   3.0  1.0  0.0       11.0  35.0    24.0        0.0   
24      UA05  2022-10-27   3.0  1.0  0.0       13.0  36.0    24.0        0.0   
25      UA05  2022-11-10   3.0  1.0  0.0       13.0  36.0    25.0        0.0   
26      UA05  2022-11-25   3.0  1.0  0.0       13.0  36.0    26.0        0.0   
27      UA05  2022-12-08   3.0  1.0  0.0       13.0  36.0    27.0        0.0   
28      UA05  2022-12-22   2.0  1.0  0.0       13.0  36.0    33.0        0.0   
29      UA05  2023-01-05   3.0  1.0  1.0       17.0  39.0    33.0        0.0   
30      UA07  2022-04-28   1.0  0.0  0.0        1.0   5.0     6.0        0.0   
31      UA07  2022-05-06   1.0  0.0  0.0        1.0   6.0     7.0        0.0   
32      UA07  2022-05-12   1.0  0.0  0.0        1.0   8.0     7.0        0.0   
33      UA07  2022-05-19   1.0  0.0  0.0        2.0   9.0     8.0        0.0   
34      UA07  2022-05-26   0.0  0.0  0.0        2.0   9.0     7.0        0.0   

     MPC  ...  SHELTER  PC_CP  PC_GBV  PC_MA  WASH  TOTAL  PROTECTIONTOTAL  \
20   6.0  ...      7.0    4.0     3.0    3.0   9.0   58.0              0.0   
21  34.0  ...     46.0   36.0    14.0    8.0  30.0  214.0              0.0   
22  34.0  ...     46.0   36.0    14.0    8.0  30.0  214.0              0.0   
23  17.0  ...     23.0   25.0    16.0    5.0  13.0  123.0              0.0   
24  17.0  ...     23.0   25.0    10.0    5.0  13.0  119.0              0.0   
25  17.3  ...     23.3   25.3    10.3    5.7  13.0  121.0              0.0   
26  17.7  ...     23.7   25.7    10.7    6.3  13.0  123.0              0.0   
27  18.0  ...     24.0   26.0    11.0    7.0  13.0  125.0              0.0   
28  18.0  ...     24.0   29.0    11.0    8.0  13.0  130.0              0.0   
29  19.0  ...     25.0   33.0    10.0    8.0  14.0  138.0              0.0   
30   8.0  ...      4.0    3.0     3.0    2.0   4.0   22.0             11.0   
31   7.0  ...      4.0    4.0     4.0    2.0   4.0   23.0              0.0   
32   8.0  ...      6.0    6.0     4.0    2.0   5.0   29.0             15.0   
33   8.0  ...      6.0    6.0     4.0    2.0   5.0   30.0             15.0   
34   7.0  ...      7.0    7.0     4.0    4.0   5.0   31.0             17.0   

    PC_PC     OBLAST PEOPLE_REACHED  
20    0.0  Vinnytska       395394.0  
21    0.0  Vinnytska       790788.0  
22    0.0  Vinnytska       790788.0  
23    0.0  Vinnytska       395394.0  
24    0.0  Vinnytska       395394.0  
25    0.0  Vinnytska       413336.0  
26    0.0  Vinnytska       431278.0  
27    0.0  Vinnytska       449220.0  
28    0.0  Vinnytska       449220.0  
29    0.0  Vinnytska       449220.0  
30    6.0   Volynska        85800.0  
31    6.0   Volynska        85800.0  
32    9.0   Volynska        85900.0  
33    9.0   Volynska        86900.0  
34    7.0   Volynska        86900.0  

[15 rows x 21 columns]

In [29]:
interpolated_df.sort_values(by=['DATE'])


ADMIN1_ID        DATE  CCCM  CCS  ETC  EDUCATION    FSL  HEALTH  \
0        UA05  2022-04-28   2.0  1.0  1.0        5.0   17.0    16.0   
690      UA74  2022-04-28   1.0  0.0  0.0        1.0   17.0    12.0   
60       UA12  2022-04-28   1.0  1.0  1.0        3.0   31.0    19.0   
660      UA73  2022-04-28   2.0  1.0  0.0        3.0   22.0     9.0   
90       UA14  2022-04-28   0.0  1.0  0.0        2.0   48.0    26.0   
..        ...         ...   ...  ...  ...        ...    ...     ...   
359      UA46  2023-01-05   4.0  1.0  1.0       23.0   62.0    52.0   
569      UA63  2023-01-05   3.0  0.0  1.0       13.0  111.0    49.0   
719      UA74  2023-01-05   0.0  0.0  0.0       18.0   56.0    29.0   
89       UA12  2023-01-05   3.0  1.0  1.0       19.0  113.0    52.0   
749      UA80  2023-01-05   0.0  1.0  1.0       18.0   45.0    63.0   

     LOGISTICS   MPC  ...  SHELTER  PC_CP  PC_GBV  PC_MA  WASH  TOTAL  \
0          0.0  10.0  ...      6.0    2.0     2.0    2.0   7.0   42.0   
690        0.0   6.0  ...      2.0    1.0     0.0    1.0  10.0   36.0   
60         0.0  12.0  ...     10.0    5.0     4.0    2.0  15.0   58.0   
660        0.0  10.0  ...      6.0    5.0     2.0    2.0  11.0   50.0   
90         0.0  12.0  ...      9.0    7.0     4.0    1.0  18.0   75.0   
..         ...   ...  ...      ...    ...     ...    ...   ...    ...   
359        1.0  26.0  ...     27.0   41.0    20.0    8.0  23.0  194.0   
569        0.0  24.0  ...     33.0   26.0     7.0   10.0  25.0  175.0   
719        0.0  20.0  ...     33.0   16.0     7.0    9.0  17.0  149.0   
89         1.0  33.0  ...     51.0   40.0    12.0    9.0  32.0  244.0   
749        1.0  26.0  ...     13.0   31.0    16.0   10.0  15.0  167.0   

     PROTECTIONTOTAL  PC_PC           OBLAST PEOPLE_REACHED  
0               16.0   11.0        Vinnytska        91200.0  
690              9.0    8.0     Chernihivska       171600.0  
60              21.0   14.0  Dnipropetrovska       231800.0  
660             17.0   12.0     Chernivetska        31200.0  
90              29.0   21.0         Donetska       122100.0  
..               ...    ...              ...            ...  
359              0.0    0.0          Lvivska       957877.0  
569              0.0    0.0       Kharkivska      2070173.0  
719              0.0    0.0     Chernihivska       477499.0  
89               0.0    0.0  Dnipropetrovska       847604.0  
749              0.0    0.0             Kyiv       761036.0  

[750 rows x 21 columns]

In [30]:
interpolated_df.describe()


CCCM         CCS         ETC   EDUCATION         FSL      HEALTH  \
count  750.000000  750.000000  750.000000  750.000000  750.000000  750.000000   
mean     0.908000    0.381333    0.192000    6.189333   34.133333   19.610667   
std      1.287016    0.538180    0.420363    5.379970   27.577556   14.599793   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    2.000000   18.000000   10.000000   
50%      0.000000    0.000000    0.000000    5.000000   29.000000   16.000000   
75%      1.000000    1.000000    0.000000    9.000000   44.000000   25.000000   
max      8.000000    2.000000    2.000000   32.000000  202.000000  100.000000   

        LOGISTICS         MPC   NUTRITION     SHELTER       PC_CP      PC_GBV  \
count  750.000000  750.000000  750.000000  750.000000  750.000000  750.000000   
mean     0.233333   13.705333    1.602667   12.088000   11.078667    5.553333   
std      0.465307    9.082728    1.544876    9.815506    9.966830    4.454329   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    9.000000    0.000000    6.000000    4.000000    2.000000   
50%      0.000000   12.000000    1.000000   10.000000    8.000000    5.000000   
75%      0.000000   18.000000    2.000000   16.000000   16.000000    8.000000   
max      2.000000   56.000000   10.000000   80.000000   52.000000   30.000000   

            PC_MA        WASH       TOTAL  PROTECTIONTOTAL       PC_PC  \
count  750.000000  750.000000  750.000000       750.000000  750.000000   
mean     4.140000   12.689333   77.418667         5.958667    1.804000   
std      2.705994    8.739412   53.084492         9.210034    4.586243   
min      0.000000    0.000000    0.000000         0.000000    0.000000   
25%      2.000000    7.000000   42.000000         0.000000    0.000000   
50%      4.000000   12.000000   66.000000         0.000000    0.000000   
75%      5.525000   16.000000  101.000000        13.000000    0.000000   
max     16.000000   58.000000  364.000000        42.000000   28.000000   

       PEOPLE_REACHED  
count    7.500000e+02  
mean     4.709115e+05  
std      4.379627e+05  
min      1.380000e+04  
25%      2.179775e+05  
50%      3.837110e+05  
75%      5.839760e+05  
max      4.140346e+06

In [31]:
# create label encoder instance
label_encoder = LabelEncoder()
#fit label encoder to transform Oblast column
encoded_oblast = label_encoder.fit_transform(interpolated_df['OBLAST'])

#create copy
df_encoded = interpolated_df.copy()

#Add column and assign values
df_encoded['OBLAST_ENCODED'] = encoded_oblast


In [32]:
# create label encoder instance
label_encoder = LabelEncoder()

#fit label encoder to transform Date column
encoded_oblast = label_encoder.fit_transform(df_encoded['DATE'])

#create copy
df_encode = df_encoded.copy()

#Add column and assign values
df_encode['DATE_ENCODED'] = encoded_oblast


In [33]:
df_encode.sort_values(by=['DATE'])


ADMIN1_ID        DATE  CCCM  CCS  ETC  EDUCATION    FSL  HEALTH  \
0        UA05  2022-04-28   2.0  1.0  1.0        5.0   17.0    16.0   
690      UA74  2022-04-28   1.0  0.0  0.0        1.0   17.0    12.0   
60       UA12  2022-04-28   1.0  1.0  1.0        3.0   31.0    19.0   
660      UA73  2022-04-28   2.0  1.0  0.0        3.0   22.0     9.0   
90       UA14  2022-04-28   0.0  1.0  0.0        2.0   48.0    26.0   
..        ...         ...   ...  ...  ...        ...    ...     ...   
359      UA46  2023-01-05   4.0  1.0  1.0       23.0   62.0    52.0   
569      UA63  2023-01-05   3.0  0.0  1.0       13.0  111.0    49.0   
719      UA74  2023-01-05   0.0  0.0  0.0       18.0   56.0    29.0   
89       UA12  2023-01-05   3.0  1.0  1.0       19.0  113.0    52.0   
749      UA80  2023-01-05   0.0  1.0  1.0       18.0   45.0    63.0   

     LOGISTICS   MPC  ...  PC_GBV  PC_MA  WASH  TOTAL  PROTECTIONTOTAL  PC_PC  \
0          0.0  10.0  ...     2.0    2.0   7.0   42.0             16.0   11.0   
690        0.0   6.0  ...     0.0    1.0  10.0   36.0              9.0    8.0   
60         0.0  12.0  ...     4.0    2.0  15.0   58.0             21.0   14.0   
660        0.0  10.0  ...     2.0    2.0  11.0   50.0             17.0   12.0   
90         0.0  12.0  ...     4.0    1.0  18.0   75.0             29.0   21.0   
..         ...   ...  ...     ...    ...   ...    ...              ...    ...   
359        1.0  26.0  ...    20.0    8.0  23.0  194.0              0.0    0.0   
569        0.0  24.0  ...     7.0   10.0  25.0  175.0              0.0    0.0   
719        0.0  20.0  ...     7.0    9.0  17.0  149.0              0.0    0.0   
89         1.0  33.0  ...    12.0    9.0  32.0  244.0              0.0    0.0   
749        1.0  26.0  ...    16.0   10.0  15.0  167.0              0.0    0.0   

              OBLAST  PEOPLE_REACHED  OBLAST_ENCODED DATE_ENCODED  
0          Vinnytska         91200.0              20            0  
690     Chernihivska        171600.0               1            0  
60   Dnipropetrovska        231800.0               3            0  
660     Chernivetska         31200.0               2            0  
90          Donetska        122100.0               4            0  
..               ...             ...             ...          ...  
359          Lvivska        957877.0              13           28  
569       Kharkivska       2070173.0               6           28  
719     Chernihivska        477499.0               1           28  
89   Dnipropetrovska        847604.0               3           28  
749             Kyiv        761036.0              10           28  

[750 rows x 23 columns]

In [34]:
df_encoded.columns

Index(['ADMIN1_ID', 'DATE', 'CCCM', 'CCS', 'ETC', 'EDUCATION', 'FSL', 'HEALTH',
       'LOGISTICS', 'MPC', 'NUTRITION', 'SHELTER', 'PC_CP', 'PC_GBV', 'PC_MA',
       'WASH', 'TOTAL', 'PROTECTIONTOTAL', 'PC_PC', 'OBLAST', 'PEOPLE_REACHED',
       'OBLAST_ENCODED'],
      dtype='object')

In [35]:
merged_sheet1.shape


(750, 21)

In [36]:
#People_Reached_by_Oblast
merged_sheet2.head(10)

ADMIN1_ID        DATE      FSL       CP  MINEACTION  PROTECTIONTOTAL  \
0      UA05  2022-04-28   7900.0    100.0       500.0          11900.0   
1      UA05  2022-05-06  14300.0    100.0       500.0          14300.0   
2      UA05  2022-05-12  15500.0    200.0       500.0          16300.0   
3      UA05  2022-05-19  17500.0    500.0       500.0          20300.0   
4      UA05  2022-05-26  17600.0   4400.0       700.0          25600.0   
5      UA05  2022-06-02  19000.0  21500.0       700.0          48000.0   
6      UA05  2022-06-09  19000.0  21500.0       700.0          21000.0   
7      UA05  2022-07-07  39200.0  46500.0     11800.0          94900.0   
8      UA05  2022-07-14  39200.0  35700.0     11800.0          75700.0   
9      UA05  2022-07-21  39200.0  35700.0     99800.0         174700.0   

    HEALTH       MPC  SHELTER     WASH  PEOPLEREACHED  GENERALPROTECTION  \
0   4600.0   91200.0  15800.0   2300.0        91200.0            11200.0   
1   4600.0  113000.0  16200.0   2300.0       113000.0             9600.0   
2   9900.0  138800.0  17500.0   3700.0       138800.0            11400.0   
3  23200.0  140600.0  17500.0   3700.0       140600.0                0.0   
4  23300.0  216600.0  21700.0   3800.0       216600.0                0.0   
5  23300.0  236000.0  33300.0   3800.0       236000.0                0.0   
6  23300.0  240100.0  40500.0  23800.0       240100.0                0.0   
7  27500.0  214600.0  59400.0  42700.0       214600.0                0.0   
8  27500.0  216000.0  60700.0  42700.0       216000.0                0.0   
9  39000.0  217200.0  62600.0  42700.0       217200.0                0.0   

      OBLAST  
0  Vinnytska  
1  Vinnytska  
2  Vinnytska  
3  Vinnytska  
4  Vinnytska  
5  Vinnytska  
6  Vinnytska  
7  Vinnytska  
8  Vinnytska  
9  Vinnytska

In [37]:
#UDE_Inputs
merged_sheet3.head()

ADMIN1_ID        DATE  PEOPLEREACHED  NUMBEROFORGANISATIONS     OBLAST
0      UA05  2022-04-28        91200.0                     42  Vinnytska
1      UA05  2022-05-06       113000.0                     45  Vinnytska
2      UA05  2022-05-12       138800.0                     49  Vinnytska
3      UA05  2022-05-19       140600.0                     50  Vinnytska
4      UA05  2022-05-26       216600.0                     48  Vinnytska

In [38]:
# copy_df = pd.DataFrame(np.nan, index=merged_sheet1.index,columns=merged_sheet1.columns)
# copy_df['DATE'] = merged_sheet1['DATE']
# copy_df['OBLAST'] = merged_sheet1['OBLAST']
# copy_df.head()

In [39]:
# # Create an Excel writer object
# excel_writer1 = pd.ExcelWriter('Data_ Round 33 - Ukraine 5W - 2022-11-10.xlsx', engine='xlsxwriter')
# excel_writer2 = pd.ExcelWriter('Data_ Round 34 - Ukraine 5W - 2022-11-25.xlsx', engine='xlsxwriter')
# # Write each DataFrame to a separate sheet in the Excel file
# copy_df.to_excel(excel_writer1, sheet_name='Num_of_Orgs_by_Oblast', index=False)
# copy_df.to_excel(excel_writer2, sheet_name='Num_of_Orgs_by_Oblast', index=False)


In [40]:
merged_sheet1.shape


(750, 21)

In [41]:
merged_sheet3.shape

(750, 5)

In [42]:
merged_sheet1.DATE.value_counts()


2022-09-29    50
2022-04-28    25
2022-08-11    25
2022-12-22    25
2022-12-08    25
2022-11-25    25
2022-11-10    25
2022-10-27    25
2022-10-13    25
2022-09-22    25
2022-09-15    25
2022-09-08    25
2022-08-25    25
2022-08-18    25
2022-08-04    25
2022-05-06    25
2022-07-28    25
2022-07-21    25
2022-07-14    25
2022-07-07    25
2022-06-30    25
2022-06-23    25
2022-06-16    25
2022-06-09    25
2022-06-02    25
2022-05-26    25
2022-05-19    25
2022-05-12    25
2023-01-05    25
Name: DATE, dtype: int64


personally working on
- PC_CP, PC_GBV, PC_MA

linear regression model (using df_encode) on the labels



In [43]:
df_encode.columns

Index(['ADMIN1_ID', 'DATE', 'CCCM', 'CCS', 'ETC', 'EDUCATION', 'FSL', 'HEALTH',
       'LOGISTICS', 'MPC', 'NUTRITION', 'SHELTER', 'PC_CP', 'PC_GBV', 'PC_MA',
       'WASH', 'TOTAL', 'PROTECTIONTOTAL', 'PC_PC', 'OBLAST', 'PEOPLE_REACHED',
       'OBLAST_ENCODED', 'DATE_ENCODED'],
      dtype='object')

# **Model for PC_CP**

In [44]:
#PC_CP model
#label is the need (y)
#
y_pccp = df_encode['PC_CP']
X_pccp = df_encode.drop(columns = ['PC_CP', 'ADMIN1_ID','DATE','OBLAST'])



In [45]:
X_train, X_test, y_train, y_test = train_test_split(X_pccp, y_pccp, test_size=0.10, random_state=1234)


In [46]:
# Create the  LinearRegression model object
model_pccp = LinearRegression()

# Fit the model to the training data
model_pccp.fit(X_train, y_train)

#  Make predictions on the test data
prediction_pccp = model_pccp.predict(X_test)

In [47]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction_pccp)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction_pccp))


Model Performance

RMSE =   3.09
 R^2 =   0.93


RMSE: Means that, by average, predictions are off by 3.09 units.



R^2: 93% of the variation of the feature was explained by model of X

In [48]:
df_encode.columns

Index(['ADMIN1_ID', 'DATE', 'CCCM', 'CCS', 'ETC', 'EDUCATION', 'FSL', 'HEALTH',
       'LOGISTICS', 'MPC', 'NUTRITION', 'SHELTER', 'PC_CP', 'PC_GBV', 'PC_MA',
       'WASH', 'TOTAL', 'PROTECTIONTOTAL', 'PC_PC', 'OBLAST', 'PEOPLE_REACHED',
       'OBLAST_ENCODED', 'DATE_ENCODED'],
      dtype='object')

In [49]:
#exclude = ['PC_CP', 'ADMIN1_ID','DATE','OBLAST']
df_encode.corr()['PC_CP'].sort_values(ascending=False)


<ipython-input-49-c99bd436bd50>:2: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df_encode.corr()['PC_CP'].sort_values(ascending=False)


PC_CP              1.000000
PC_GBV             0.908628
TOTAL              0.887108
EDUCATION          0.875966
SHELTER            0.830870
MPC                0.821972
PC_MA              0.795660
WASH               0.728580
HEALTH             0.709541
FSL                0.708309
CCCM               0.651496
NUTRITION          0.538245
DATE_ENCODED       0.519384
CCS                0.410320
LOGISTICS          0.357334
PEOPLE_REACHED     0.339911
ETC                0.301036
OBLAST_ENCODED     0.000037
PC_PC             -0.224915
PROTECTIONTOTAL   -0.264239
Name: PC_CP, dtype: float64

In [50]:
y_pccp = df_encode['PC_CP']
X_pccp = df_encode.drop(columns = ['PC_CP', 'ADMIN1_ID','DATE','OBLAST','CCS','LOGISTICS','PEOPLE_REACHED',
                              'ETC', 'OBLAST_ENCODED', 'PC_PC', 'PROTECTIONTOTAL'])

X_train, X_test, y_train, y_test = train_test_split(X_pccp, y_pccp, test_size=0.10, random_state=1234)
# Create the  LinearRegression model object
model_pccp = LinearRegression()

# Fit the model to the training data
model_pccp.fit(X_train, y_train)

#  Make predictions on the test data
prediction_pccp = model_pccp.predict(X_test)

In [51]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction_pccp)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction_pccp))


Model Performance

RMSE =   3.10
 R^2 =   0.93


# **Model for PC_GBV**

In [52]:
#PC_GBV model
#label is the need (y)

#encluding any features with <0.5 corr
y_pcgbv = df_encode['PC_GBV']
X_pcgbv = df_encode.drop(columns = ['PC_GBV', 'ADMIN1_ID','DATE','OBLAST'])



In [53]:
X_train, X_test, y_train, y_test = train_test_split(X_pcgbv, y_pcgbv, test_size=0.10, random_state=1234)


In [54]:
# Create the  LinearRegression model object
model_pcgbv = LinearRegression()

# Fit the model to the training data
model_pcgbv.fit(X_train, y_train)

#  Make predictions on the test data
prediction = model_pcgbv.predict(X_test)

In [55]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction))


Model Performance

RMSE =   1.47
 R^2 =   0.93


RMSE: Means that, by average, predictions are off by 1.47 units.



R^2: 93% of the variation of the feature was explained by model of X

In [56]:
df_encode.corr()['PC_GBV'].sort_values(ascending=False)


<ipython-input-56-3d6fe0552d06>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df_encode.corr()['PC_GBV'].sort_values(ascending=False)


PC_GBV             1.000000
PC_CP              0.908628
EDUCATION          0.864020
TOTAL              0.842423
SHELTER            0.750290
MPC                0.746366
PC_MA              0.741363
HEALTH             0.739086
WASH               0.683704
CCCM               0.646779
FSL                0.639221
NUTRITION          0.563990
LOGISTICS          0.438138
DATE_ENCODED       0.437990
CCS                0.400299
ETC                0.361024
PEOPLE_REACHED     0.293237
OBLAST_ENCODED     0.075641
PC_PC             -0.185717
PROTECTIONTOTAL   -0.227447
Name: PC_GBV, dtype: float64

In [57]:
y_pcgbv = df_encode['PC_GBV']
X_pcgbv = df_encode.drop(columns = ['PC_GBV', 'ADMIN1_ID','DATE','OBLAST','LOGISTICS', 'DATE_ENCODED','CCS','ETC','PEOPLE_REACHED',
                                    'OBLAST_ENCODED','PC_PC','PROTECTIONTOTAL'])

X_train, X_test, y_train, y_test = train_test_split(X_pcgbv, y_pcgbv, test_size=0.10, random_state=1234)
# Create the  LinearRegression model object
model_pcgbv = LinearRegression()

# Fit the model to the training data
model_pcgbv.fit(X_train, y_train)

#  Make predictions on the test data
prediction_pcgbv = model_pcgbv.predict(X_test)

In [58]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction_pcgbv)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction_pcgbv))


Model Performance

RMSE =   1.53
 R^2 =   0.92


# **Model for PC_MA**

In [59]:
#PC_MA model
#label is the need (y)
#
y_pcma = df_encode['PC_MA']
X_pcma = df_encode.drop(columns = ['PC_MA', 'ADMIN1_ID','DATE','OBLAST'])



In [60]:
X_train, X_test, y_train, y_test = train_test_split(X_pcma, y_pcma, test_size=0.10, random_state=1234)


In [61]:
# Create the  LinearRegression model object
model_pcma = LinearRegression()

# Fit the model to the training data
model_pcma.fit(X_train, y_train)

#  Make predictions on the test data
prediction = model_pcma.predict(X_test)

In [62]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction))


Model Performance

RMSE =   1.09
 R^2 =   0.88


RMSE: Means that, by average, predictions are off by 1.09 units.



R^2: 88% of the variation of the feature was explained by model of X

In [63]:
df_encode.corr()['PC_MA'].sort_values(ascending=False)


<ipython-input-63-3514bde73201>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df_encode.corr()['PC_MA'].sort_values(ascending=False)


PC_MA              1.000000
TOTAL              0.857679
MPC                0.830259
EDUCATION          0.818421
PC_CP              0.795660
SHELTER            0.746351
PC_GBV             0.741363
FSL                0.704889
HEALTH             0.681301
WASH               0.655205
DATE_ENCODED       0.536222
NUTRITION          0.485357
PEOPLE_REACHED     0.404632
CCCM               0.399715
LOGISTICS          0.283645
CCS                0.234658
ETC                0.214604
OBLAST_ENCODED    -0.079042
PC_PC             -0.287716
PROTECTIONTOTAL   -0.360783
Name: PC_MA, dtype: float64

In [64]:
y_pcma = df_encode['PC_MA']
X_pcma = df_encode.drop(columns = ['PC_MA', 'ADMIN1_ID','DATE','OBLAST','NUTRITION','PEOPLE_REACHED','CCCM','LOGISTICS','CCS','ETC','OBLAST_ENCODED',
                                    'PC_PC', 'PROTECTIONTOTAL'])

X_train, X_test, y_train, y_test = train_test_split(X_pcma, y_pcma, test_size=0.10, random_state=1234)
# Create the  LinearRegression model object
model_pcma = LinearRegression()

# Fit the model to the training data
model_pcma.fit(X_train, y_train)

#  Make predictions on the test data
prediction_pcma = model_pcma.predict(X_test)

In [65]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction_pcma)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction_pcma))


Model Performance

RMSE =   1.25
 R^2 =   0.84
